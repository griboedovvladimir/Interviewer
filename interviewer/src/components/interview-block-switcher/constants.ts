export const BLOCK_NAME_HTML =  'HTML';
export const BLOCK_NAME_CSS =  'CSS';
export const BLOCK_NAME_JS =  'JS';

export const BLOCK_ID_HTML = 'HTMLSwitcherBlock';
export const BLOCK_ID_CSS = 'CSSSwitcherBlock';
export const BLOCK_ID_JS = 'JSSwitcherBlock';

export const BLOCK_ACTIVE_CLASS = 'question-block-active';