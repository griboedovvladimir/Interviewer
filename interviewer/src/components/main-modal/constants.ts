export const ACTION_ADD_INTERVIEW = 'ADD_INTERVIEW';