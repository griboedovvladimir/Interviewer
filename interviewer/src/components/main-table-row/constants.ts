export const ACTION_REMOVE_INTERVIEW = 'REMOVE_INTERVIEW';
export const ACTION_GET_INTERVIEW = 'GET_INTERVIEW';
export const ACTION_ACTIVATE_ROW = 'ACTIVATE_ROW';
