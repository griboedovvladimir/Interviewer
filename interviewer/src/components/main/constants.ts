export const ACTION_ADD_INTERVIEW = 'ADD_INTERVIEW';
export const ACTION_REMOVE_INTERVIEW = 'REMOVE_INTERVIEW';
export const ACTION_GET_INTERVIEW = 'GET_INTERVIEW';