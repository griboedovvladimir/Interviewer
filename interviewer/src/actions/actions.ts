import * as CONSTANTS from '../constants';

export function login(logged: string) {
    return {
        type: CONSTANTS.ACTION_LOGIN,
        logged
    };
}

export function addInterview(name: string, level: string, specialization: string, id: string) {
    return {
        type: CONSTANTS.ACTION_ADD_INTERVIEW,
        id, name, level, specialization
    };
}

export function removeInterview(id: string) {
    return {
        type: CONSTANTS.ACTION_REMOVE_INTERVIEW,
        id
    };
}
