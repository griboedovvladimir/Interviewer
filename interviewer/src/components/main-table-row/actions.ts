import * as CONSTANTS from './constants';

export function removeInterview(id: string) {
    return {
        type: CONSTANTS.ACTION_REMOVE_INTERVIEW,
        id
    };
}

export function activateRow(rowData: any) {
    return {
        type: CONSTANTS.ACTION_ACTIVATE_ROW,
        rowData
    };
}