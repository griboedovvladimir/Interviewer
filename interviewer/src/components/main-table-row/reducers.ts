import * as CONSTANTS from './constants';

export function row(state = {}, action: any){
    if(action.type === CONSTANTS.ACTION_ACTIVATE_ROW){
        return {...action.rowData}
    }
    return state
}