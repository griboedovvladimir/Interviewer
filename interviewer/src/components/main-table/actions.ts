import * as CONSTANTS from "../main-table-row/constants";

export function getInterview(interviews: any) {
    return {
        type: CONSTANTS.ACTION_GET_INTERVIEW,
        interviews
    };
}

