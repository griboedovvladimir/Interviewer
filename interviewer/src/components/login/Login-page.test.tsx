import * as React from 'react';
import {mount} from 'enzyme';
import * as actions from './actions';
import * as CONSTANTS from './constants';
import {login} from "./reducers";
import "isomorphic-fetch";

const logged = true;
/*-------------------------- Redux tests -------------------------------*/
describe('login actions', () => {
    it('should change boolean value logged', () => {
        const expectedAction = {
            type: CONSTANTS.ACTION_LOGIN,
            logged
        };
        expect(actions.login(logged)).toEqual(expectedAction)
    })
});

describe('login reducers', () => {
    it('should handle ACTION_LOGIN', () => {
        expect(
            login({logged: false}, {
                type: CONSTANTS.ACTION_LOGIN,
                logged: true
            })
        ).toEqual({logged: true});

        expect(
            login({logged: false}, {
                type: CONSTANTS.ACTION_LOGIN,
                logged: true
            })
        ).toEqual({logged: true});

        expect(
            login(undefined, {
                type: CONSTANTS.ACTION_LOGIN,
                logged: true
            })
        ).toEqual({logged: true});

        expect(
            login({logged: false}, {
                type: '',
                logged: true
            })
        ).toEqual({logged: false})

    });

    it('should handle ACTION_LOGOUT', () => {
        expect(
            login({logged: false}, {
                type: CONSTANTS.ACTION_LOGOUT,
                logged: true
            })
        ).toEqual({logged: true});

        expect(
            login({logged: false}, {
                type: CONSTANTS.ACTION_LOGIN,
                logged: true
            })
        ).toEqual({logged: true});

        expect(
            login({logged: true}, {
                type: '',
                logged: false
            })
        ).toEqual({logged: true})
    });


});
/*----------------------- Components tests----------------------------*/
import ConnectedLoginPage from './Login-page';
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";
import {BrowserRouter} from "react-router-dom";

let store: any, wrapper: any;
let initialState = {login: {logged: true}};
let mockStore = configureStore();

describe("<LoginPage />  UI Component", () => {
    beforeEach(() => {
        store = mockStore(initialState);
        wrapper = mount(<Provider store={store}><BrowserRouter><ConnectedLoginPage/></BrowserRouter></Provider>)
    });

    it("render default login-page", () => {
        expect(wrapper.find(ConnectedLoginPage).length).toEqual(1)
        expect(wrapper).toMatchSnapshot();
    });

});

initialState = {login: {logged: false}};
mockStore = configureStore();
describe("<LoginPage />  UI Component", () => {
    beforeEach(() => {
        store = mockStore(initialState);
        wrapper = mount(<Provider store={store}><BrowserRouter><ConnectedLoginPage/></BrowserRouter></Provider>)
    });

    it("render default login-page wit logged === false", () => {
        expect(wrapper.find(ConnectedLoginPage).length).toEqual(1)
        expect(wrapper).toMatchSnapshot();
    });

    it("when clicking the login-btn", () => {
        expect(wrapper).toMatchSnapshot();
        beforeEach(() => {
            wrapper.find('[name="login-btn"]').simulate('click', {
                preventDefault: () => {
                },
            })
        });
        expect(wrapper).toMatchSnapshot();
    });

    it("when submitting the form", () => {
        expect(wrapper).toMatchSnapshot();
        beforeEach(() => {
            wrapper.find('form').simulate('submit', {
                preventDefault: () => {
                },
            })
        });
    });
    expect(wrapper).toMatchSnapshot();
});




