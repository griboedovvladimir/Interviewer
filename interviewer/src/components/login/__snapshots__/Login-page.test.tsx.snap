// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `undefined`;

exports[`<LoginPage />  UI Component render default login-page 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(LoginPage)>
        <LoginPage
          action={
            Object {
              "login": [Function],
            }
          }
          logged={false}
        >
          <div
            className="form-wrapper"
          >
            <form
              className="login-form"
              onSubmit={[Function]}
            >
              <div
                className="mdc-component mdc-component__textfield"
              >
                <div
                  className="mdc-component__containers"
                  dir-horizontal=""
                >
                  <div
                    className="mdc-component__section mdc-component__section--size-narrow"
                  >
                    <div
                      className="mdc-component__containers__primary mdc-textfield--theme-primary"
                    >
                      <div
                        className="mdc-textfield"
                      >
                        <input
                          className="mdc-textfield__input"
                          id="loginName"
                          minLength={3}
                          name="loginName"
                          pattern="[A-z.]+"
                          placeholder="Your name"
                          required={true}
                          type="text"
                        />
                        <label
                          className="mdc-textfield__label mdc-textfield__label--float-above"
                          htmlFor="loginName"
                        >
                          Name
                        </label>
                      </div>
                    </div>
                    <div
                      className="mdc-component__containers__secondary mdc-textfield--theme-primary"
                    >
                      <div
                        className="mdc-textfield"
                      >
                        <input
                          className="mdc-textfield__input"
                          id="loginPassword"
                          name="loginPassword"
                          placeholder="Your password"
                          required={true}
                          type="password"
                        />
                        <label
                          className="mdc-textfield__label mdc-textfield__label--float-above"
                          htmlFor="loginPassword"
                        >
                          Passsword
                        </label>
                      </div>
                    </div>
                    <div
                      className="mdc-component mdc-component__buttons"
                    >
                      <div
                        className="mdc-component__containers__primary"
                      >
                        <button
                          className="mdc-button mdc-button--raised"
                          name="login-btn"
                          type="submit"
                        >
                          Submit
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </form>
          </div>
        </LoginPage>
      </Connect(LoginPage)>
    </Router>
  </BrowserRouter>
</Provider>
`;

exports[`<LoginPage />  UI Component render default login-page wit logged === false 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(LoginPage)>
        <LoginPage
          action={
            Object {
              "login": [Function],
            }
          }
          logged={false}
        >
          <div
            className="form-wrapper"
          >
            <form
              className="login-form"
              onSubmit={[Function]}
            >
              <div
                className="mdc-component mdc-component__textfield"
              >
                <div
                  className="mdc-component__containers"
                  dir-horizontal=""
                >
                  <div
                    className="mdc-component__section mdc-component__section--size-narrow"
                  >
                    <div
                      className="mdc-component__containers__primary mdc-textfield--theme-primary"
                    >
                      <div
                        className="mdc-textfield"
                      >
                        <input
                          className="mdc-textfield__input"
                          id="loginName"
                          minLength={3}
                          name="loginName"
                          pattern="[A-z.]+"
                          placeholder="Your name"
                          required={true}
                          type="text"
                        />
                        <label
                          className="mdc-textfield__label mdc-textfield__label--float-above"
                          htmlFor="loginName"
                        >
                          Name
                        </label>
                      </div>
                    </div>
                    <div
                      className="mdc-component__containers__secondary mdc-textfield--theme-primary"
                    >
                      <div
                        className="mdc-textfield"
                      >
                        <input
                          className="mdc-textfield__input"
                          id="loginPassword"
                          name="loginPassword"
                          placeholder="Your password"
                          required={true}
                          type="password"
                        />
                        <label
                          className="mdc-textfield__label mdc-textfield__label--float-above"
                          htmlFor="loginPassword"
                        >
                          Passsword
                        </label>
                      </div>
                    </div>
                    <div
                      className="mdc-component mdc-component__buttons"
                    >
                      <div
                        className="mdc-component__containers__primary"
                      >
                        <button
                          className="mdc-button mdc-button--raised"
                          name="login-btn"
                          type="submit"
                        >
                          Submit
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </form>
          </div>
        </LoginPage>
      </Connect(LoginPage)>
    </Router>
  </BrowserRouter>
</Provider>
`;

exports[`<LoginPage />  UI Component when clicking the login-btn 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(LoginPage)>
        <LoginPage
          action={
            Object {
              "login": [Function],
            }
          }
          logged={false}
        >
          <div
            className="form-wrapper"
          >
            <form
              className="login-form"
              onSubmit={[Function]}
            >
              <div
                className="mdc-component mdc-component__textfield"
              >
                <div
                  className="mdc-component__containers"
                  dir-horizontal=""
                >
                  <div
                    className="mdc-component__section mdc-component__section--size-narrow"
                  >
                    <div
                      className="mdc-component__containers__primary mdc-textfield--theme-primary"
                    >
                      <div
                        className="mdc-textfield"
                      >
                        <input
                          className="mdc-textfield__input"
                          id="loginName"
                          minLength={3}
                          name="loginName"
                          pattern="[A-z.]+"
                          placeholder="Your name"
                          required={true}
                          type="text"
                        />
                        <label
                          className="mdc-textfield__label mdc-textfield__label--float-above"
                          htmlFor="loginName"
                        >
                          Name
                        </label>
                      </div>
                    </div>
                    <div
                      className="mdc-component__containers__secondary mdc-textfield--theme-primary"
                    >
                      <div
                        className="mdc-textfield"
                      >
                        <input
                          className="mdc-textfield__input"
                          id="loginPassword"
                          name="loginPassword"
                          placeholder="Your password"
                          required={true}
                          type="password"
                        />
                        <label
                          className="mdc-textfield__label mdc-textfield__label--float-above"
                          htmlFor="loginPassword"
                        >
                          Passsword
                        </label>
                      </div>
                    </div>
                    <div
                      className="mdc-component mdc-component__buttons"
                    >
                      <div
                        className="mdc-component__containers__primary"
                      >
                        <button
                          className="mdc-button mdc-button--raised"
                          name="login-btn"
                          type="submit"
                        >
                          Submit
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </form>
          </div>
        </LoginPage>
      </Connect(LoginPage)>
    </Router>
  </BrowserRouter>
</Provider>
`;

exports[`<LoginPage />  UI Component when clicking the login-btn 2`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(LoginPage)>
        <LoginPage
          action={
            Object {
              "login": [Function],
            }
          }
          logged={false}
        >
          <div
            className="form-wrapper"
          >
            <form
              className="login-form"
              onSubmit={[Function]}
            >
              <div
                className="mdc-component mdc-component__textfield"
              >
                <div
                  className="mdc-component__containers"
                  dir-horizontal=""
                >
                  <div
                    className="mdc-component__section mdc-component__section--size-narrow"
                  >
                    <div
                      className="mdc-component__containers__primary mdc-textfield--theme-primary"
                    >
                      <div
                        className="mdc-textfield"
                      >
                        <input
                          className="mdc-textfield__input"
                          id="loginName"
                          minLength={3}
                          name="loginName"
                          pattern="[A-z.]+"
                          placeholder="Your name"
                          required={true}
                          type="text"
                        />
                        <label
                          className="mdc-textfield__label mdc-textfield__label--float-above"
                          htmlFor="loginName"
                        >
                          Name
                        </label>
                      </div>
                    </div>
                    <div
                      className="mdc-component__containers__secondary mdc-textfield--theme-primary"
                    >
                      <div
                        className="mdc-textfield"
                      >
                        <input
                          className="mdc-textfield__input"
                          id="loginPassword"
                          name="loginPassword"
                          placeholder="Your password"
                          required={true}
                          type="password"
                        />
                        <label
                          className="mdc-textfield__label mdc-textfield__label--float-above"
                          htmlFor="loginPassword"
                        >
                          Passsword
                        </label>
                      </div>
                    </div>
                    <div
                      className="mdc-component mdc-component__buttons"
                    >
                      <div
                        className="mdc-component__containers__primary"
                      >
                        <button
                          className="mdc-button mdc-button--raised"
                          name="login-btn"
                          type="submit"
                        >
                          Submit
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </form>
          </div>
        </LoginPage>
      </Connect(LoginPage)>
    </Router>
  </BrowserRouter>
</Provider>
`;

exports[`<LoginPage />  UI Component when submitting the form 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(LoginPage)>
        <LoginPage
          action={
            Object {
              "login": [Function],
            }
          }
          logged={false}
        >
          <div
            className="form-wrapper"
          >
            <form
              className="login-form"
              onSubmit={[Function]}
            >
              <div
                className="mdc-component mdc-component__textfield"
              >
                <div
                  className="mdc-component__containers"
                  dir-horizontal=""
                >
                  <div
                    className="mdc-component__section mdc-component__section--size-narrow"
                  >
                    <div
                      className="mdc-component__containers__primary mdc-textfield--theme-primary"
                    >
                      <div
                        className="mdc-textfield"
                      >
                        <input
                          className="mdc-textfield__input"
                          id="loginName"
                          minLength={3}
                          name="loginName"
                          pattern="[A-z.]+"
                          placeholder="Your name"
                          required={true}
                          type="text"
                        />
                        <label
                          className="mdc-textfield__label mdc-textfield__label--float-above"
                          htmlFor="loginName"
                        >
                          Name
                        </label>
                      </div>
                    </div>
                    <div
                      className="mdc-component__containers__secondary mdc-textfield--theme-primary"
                    >
                      <div
                        className="mdc-textfield"
                      >
                        <input
                          className="mdc-textfield__input"
                          id="loginPassword"
                          name="loginPassword"
                          placeholder="Your password"
                          required={true}
                          type="password"
                        />
                        <label
                          className="mdc-textfield__label mdc-textfield__label--float-above"
                          htmlFor="loginPassword"
                        >
                          Passsword
                        </label>
                      </div>
                    </div>
                    <div
                      className="mdc-component mdc-component__buttons"
                    >
                      <div
                        className="mdc-component__containers__primary"
                      >
                        <button
                          className="mdc-button mdc-button--raised"
                          name="login-btn"
                          type="submit"
                        >
                          Submit
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </form>
          </div>
        </LoginPage>
      </Connect(LoginPage)>
    </Router>
  </BrowserRouter>
</Provider>
`;
